base/deprecated.jl

```diff
@@ -788,6 +788,38 @@ function symperm{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, pinv::Vector{Ti})
         "Pkg.add(\"SuiteSparse\") to install SuiteSparse on Julia v0.5."))
 end
 
+# needs to be a macro so that we can use ::@mime(s) in type declarations
+eval(Multimedia, quote
+export @MIME
+macro MIME(s)
+    Base.warn_once("@MIME(\"\") is deprecated, use MIME\"\" instead.")
+    if isa(s,AbstractString)
+        :(MIME{$(Expr(:quote, Symbol(s)))})
+    else
+        :(MIME{Symbol($s)})
+    end
+end
+end)
+
+# Deprecate vectorized unary functions over sparse matrices in favor of compact broadcast syntax
+macro _enumerate_sparsemat_vectobroad_deps(fs...)
+	depset = Expr(:block)
+	for f in fs
+		push!(depset.args, :( @deprecate $(esc(f))(A::SparseMatrixCSC) $(esc(f)).(A) ) )
+	end
+	return depset
+end
+@_enumerate_sparsemat_vectobroad_deps(
```

Comment:
Do you mean a loop calling `eval` for each method? If so, sure. I defaulted to a macro having written an identical macro just before, but would be happy to change that to an `eval` loop for the deps if you prefer. If you mean something else by `loop`? Thanks!